src/fsspec_xrootd/xrootd.py

```diff
@@ -75,66 +147,96 @@ def _strip_protocol(cls, path: str | list[str]) -> Any:
         else:
             raise ValueError("Strip protocol not given string or list")
 
-    def mkdir(self, path: str, create_parents: bool = True, **kwargs: Any) -> None:
+    async def _mkdir(
+        self, path: str, create_parents: bool = True, **kwargs: Any
+    ) -> None:
         if create_parents:
-            status, n = self._myclient.mkdir(
-                path, MkDirFlags.MAKEPATH, timeout=self.timeout
+            status, n = await _async_wrap(
+                self._myclient.mkdir, path, MkDirFlags.MAKEPATH, self.timeout
             )
         else:
-            status, n = self._myclient.mkdir(path, timeout=self.timeout)
+            status, n = await _async_wrap(self._myclient.mkdir, path, self.timeout)
         if not status.ok:
             raise OSError(f"Directory not made properly: {status.message}")
 
-    def makedirs(self, path: str, exist_ok: bool = False) -> None:
+    async def _makedirs(self, path: str, exist_ok: bool = False) -> None:
         if not exist_ok:
-            if self.exists(path):
+            if await self._exists(path):
                 raise OSError(
                     "Location already exists and exist_ok arg was set to false"
                 )
-        status, n = self._myclient.mkdir(
-            path, MkDirFlags.MAKEPATH, timeout=self.timeout
+        status, n = await _async_wrap(
+            self._myclient.mkdir, path, MkDirFlags.MAKEPATH, self.timeout
         )
         if not status.ok and not (status.code == ErrorCodes.INVALID_PATH and exist_ok):
             raise OSError(f"Directory not made properly: {status.message}")
 
-    def rmdir(self, path: str) -> None:
-        status, n = self._myclient.rmdir(path, timeout=self.timeout)
+    async def _rm(
+        self,
+        path: str,
+        recursive: bool = False,
+        batch_size: int | None = None,
+        **kwargs: Any,
+    ) -> Any:
```

Comment:
Make a TODO to deal with out-of-order? Can also make a github issue here to remind us for later when https://github.com/fsspec/filesystem_spec/pull/1014 is in a release

Reply:
ok